fix(core): update core config file and handle read errors

mpsenn opened the pull request:
The core config file containing environment variables is now build/dev.properties. Use that file instead of config.blt. Additionally handle any errors in parsing and reading by logging to console and not attempting to write to the file.

Fixes [@W-13007946@](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001OplXFYAZ/view). Fixes #497.

mpsenn commented:
Fixes #497.

mpsenn commented:
Note: for git checkout of core, this generates settings with empty p4.port values, etc. This is fine, I think the eslint executable is what really matters.

randi274 reviewed (changes requested):
QA looks good! For our SFDX users, we see the language server boots up as expected, and the usual functionality like hover, go-to-definition and auto-complete work great.

I am seeing some failing tests in our internal PR, and they're reproducible locally. Likely just a stubbing issue that needs to be updated given the new path. Let me know when you're able to take a look and I'll get the updated changes merged into my branch. Thanks!

https://github.com/forcedotcom/lightning-language-server/actions/runs/4680070576/jobs/8290938179

mpsenn commented:
@randi274 I've updated the change, after taking a closer look at a colleague's setup, it looks like env. variables are used for the config now instead of a properties file. So I've updated the implementation, and modified the test workspace to match.

randi274 commented:
Thanks again @mpsenn! We'll get this updated in vs code ASAP 🎉
